apps/cookbook/src/app/showcase/modal-experimental-showcase/modal-experimental-showcase.component.ts

```diff
@@ -0,0 +1,270 @@
+    {
+      name: 'componentProps',
+      description: '(Optional) The data to pass to the modal component.',
+      defaultValue: 'undefined',
+      type: ['undefined | { [key: string]: any; }'],
+    },
+    {
+      name: 'cssClass',
+      description: `(Optional) Adds custom css classes to the modal. This allows for custom styling of the modal (see 'CSS Custom Properties' section).`,
+      defaultValue: 'undefined',
+      type: ['string | string[]'],
+    },
+    {
+      name: 'canDismiss',
+      description: `(Optional) If canDismiss is true, then the modal will close when users attempt to dismiss the modal. If canDismiss is false, then the modal will not close when users attempt to dismiss the modal.
+      
+      canDismiss can also be a function, which must return a Promise that resolves to either true or false. If the promise resolves to true, then the modal will dismiss. If the promise resolves to false, then the modal will not dismiss.
+      `,
+      defaultValue: 'undefined',
+      type: ['boolean | (() => Promise<boolean>)'],
+    },
+    {
+      name: 'backdropDismiss',
+      description: `(Optional) If true, the modal will be dismissed when the backdrop is clicked.`,
+      defaultValue: 'true',
+      type: ['boolean'],
+    },
+    {
+      name: 'showBackdrop',
+      description: `(Optional) If true, a backdrop will be displayed behind the modal. This property controls whether or not the backdrop darkens the screen when the modal is presented. It does not control whether or not the backdrop is active or present in the DOM..`,
+      defaultValue: 'true',
+      type: ['boolean'],
+    },
+  ];
+
+  wrapperComponentProperties: ApiDescriptionProperty[] = [
+    {
+      name: 'title',
+      description: `The title of the modal`,
+      defaultValue: '',
+      type: ['string'],
+    },
+    {
+      name: 'hasCollapsibleTitle',
+      description: `(Optional) If \`true\` will cause the title to initially be rendered as part of the content; once scrolled out of view it collapses and appears in the header area. 
+      \n Useful for long titles that would otherwise truncate. `,
+      defaultValue: 'true',
+      type: ['boolean'],
+    },
+    {
+      name: 'scrollDisabled',
+      description: 'Disable scrolling in the modal',
+      type: ['boolean'],
+      defaultValue: 'false',
+    },
+  ];
+
+  controllerMethods: ApiDescriptionMethod[] = [
+    {
+      name: 'showModal',
+      description:
+        'Generates and presents a modal. It takes a required argument of "ModalExperimentalConfig".',
+      signature: `{
+        onWillDismiss: Observable<OverlayEventDetail>;
+        onDidDismiss: Observable<OverlayEventDetail>;
+      }`,
+    },
+    {
+      name: 'closeModal',
+      description: `Closes the top-most modal. It takes two optional arguments: data & role.
+
+        data is the data you want to get out of the modal, by subscribing to either "onWillDismiss" or "onDidDismiss" as described below. 
```

Comment:
Data in a codebolck?

Reply:
I agree that it should be in a codeblock, but our `cookbook-api-description-methods` component doesn't support it unfortunately.